### PR TITLE
(2126) Only central Administrators and Registrars can edit a profession name and organisation

### PR DESCRIPTION
**Pull request by bryangaledxw**
### Before
localhost_3000_admin_professions_871afd7c-3951-4869-abb6-7d91cf60fcd1_versions_771973d1-f9b3-49d8-b37e-ab3e908c7a08_check-your-answers_edit=true

### After
localhost_3000_admin_professions_871afd7c-3951-4869-abb6-7d91cf60fcd1_versions_771973d1-f9b3-49d8-b37e-ab3e908c7a08_check-your-answers_edit=true (1)

### CODE REVIEW

File: views/admin/professions/check-your-answers.njk [hunk 1]
```diff
@@ -35,7 +35,7 @@
                       visuallyHiddenText: ("professions.form.label.topLevelInformation.name" | t)
                     }
                   ]
-                }
+                } if 'createProfession' in permissions else undefined
```
Comment (pezholio) [inline views/admin/professions/check-your-answers.njk:38]: Oh nice, I didn't realise you could do this in Nunjucks templates

**Review by pezholio — APPROVED** [review]
LGTM - just one thing 👍

File: src/users/helpers/check-user-is-service-owner.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { UnauthorizedException } from '@nestjs/common';
+import { RequestWithAppSession } from '../../common/interfaces/request-with-app-session.interface';
+import { User } from '../user.entity';
+
+export function checkUserIsServiceOwner(req: RequestWithAppSession): void {
```
[the remark this answers is not in the document]
Comment (bryangaledxw) [reply src/users/helpers/check-user-is-service-owner.ts:5]: I'm tempted to go the other way and change the name to `check**Acting**UserIsServiceOwner`? Then it maybe makes more sense that it's examining the request.

Thoughts?

The bit I really want to keep encapsulated in this function is where to look on the request for the acting user, especially as `appSession` has type `any`. I don't want to be doing a type-unsafe lookup in multiple places

[the remark this answers is not in the document]
Comment (bryangaledxw) [reply src/users/helpers/check-user-is-service-owner.ts:5]: Actually, gonna do a quick `getActingUser` helper

**Review by pezholio — APPROVED** [review]
Yeah, this is ace. `getActingUser` looks like it'll be a big help 👍

**Pull request merged**